src/CLMemory.ts

```diff
@@ -140,10 +150,10 @@ export class CLMemory {
     }
 
     public get BotState(): BotState {
-        return BotState.Get(this)
+        return BotState.Get(this, this.turnContext ? BB.TurnContext.getConversationReference(this.turnContext.activity) : null)
     }
 
-    public get TurnContext(): BB.TurnContext | undefined {
+    public get TurnContext(): BB.TurnContext | null {
```

Comment:
Not really a code change, but interested in your choice between `null` and `undefined`?
It seems Lars uses `null` everywhere, and in C# we don't have undefined so I wondered how you think about these?

I tend to view `null` and explicit assignment of nothing and consumers who see that value know it was put there intentionally, where as `undefined` is a default value or used to put an object back into a default state.

Reply:
Based on typescript coding guidelines, we should use `undefined` over `null` https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined

I kind of forced to change this to `null`, because of usage of `null` in underlying class, e.g. BotState. Probably a good practice to switch to `undefined` based on the guidelines.